src/frequenz/sdk/_internal/_channels.py

```diff
@@ -24,3 +25,26 @@ def new_receiver(self, maxsize: int = 50) -> Receiver[T]:
         Returns:
             A receiver instance.
         """
+
+
+class LatestValueCache(typing.Generic[T]):
+    """A cache that stores the latest value in a receiver."""
+
+    def __init__(self, receiver: Receiver[T]) -> None:
+        """Create a new cache.
+
+        Args:
+            receiver: The receiver to cache.
+        """
+        self._receiver = receiver
+        self._latest_value: T | None = None
+        self._task = asyncio.create_task(self._run())
+
+    @property
+    def latest_value(self) -> T | None:
+        """Get the latest value in the cache."""
+        return self._latest_value
```

Comment:
With that in mind, this should either return the sentinel, or maybe raise an exception?

Alternative it could be parametrized also with a second type, and receive a default at creation time, like:

```py
class LatestValueCache(typing.Generic[T, U]):
    """A cache that stores the latest value in a receiver."""

    def __init__(self, receiver: Receiver[T], *, default: U) -> None:
        ...

last = LatestValueCache[int, None](receiver, default=None)
```

Although it is a bit convoluted for the average case.

Reply:
I've got it to raise, but also added a `has_value` method that people can check first if they want to avoid having to deal with an exception: https://github.com/frequenz-floss/frequenz-sdk-python/pull/846